src/azure-cli-core/azure/cli/core/extension/dynamic_install.py

```diff
@@ -121,7 +121,14 @@ def try_install_extension(parser, args):
 
 
 def _get_extension_use_dynamic_install_config(cli_ctx):
-    default_value = 'yes_prompt'
+    from knack.prompting import verify_is_a_tty, NoTTYException
```

Comment:
Have you tested the change in both tty and non-tty environment?

Reply:
Add function tests